File: api-pytest-framework/payloads/booking_payloads.py
def create_booking(firstname="Jim", lastname="Brown", totalprice=111,
                   depositpaid=True, checkin="2025-01-01", checkout="2025-01-10",
                   additionalneeds="Breakfast") -> dict:
    return {
        "firstname":       firstname,
        "lastname":        lastname,
        "totalprice":      totalprice,
        "depositpaid":     depositpaid,
        "bookingdates":    {"checkin": checkin, "checkout": checkout},
        "additionalneeds": additionalneeds,
    }


def create_booking_without_field(field: str) -> dict:
    """Retourne un payload complet dont le champ `field` est omis."""
    full = create_booking()
    full.pop(field, None)
    return full


def create_booking_minimal() -> dict:
    """Payload sans le champ optionnel additionalneeds."""
    return {
        "firstname":    "Jim",
        "lastname":     "Brown",
        "totalprice":   111,
        "depositpaid":  True,
        "bookingdates": {"checkin": "2025-01-01", "checkout": "2025-01-10"},
    }


def update_booking(**kwargs) -> dict:
    payload = create_booking()
    payload.update(kwargs)
    return payload


def patch_booking(firstname: str = None, lastname: str = None,
                  totalprice: int = None) -> dict:
    payload = {}
    if firstname  is not None: payload["firstname"]  = firstname
    if lastname   is not None: payload["lastname"]   = lastname
    if totalprice is not None: payload["totalprice"] = totalprice
    return payload


def auth_payload(username: str = "admin", password: str = "password123") -> dict:
    return {"username": username, "password": password}

File: api-Java-Rest-Assured/src/main/java/com/restfulbooker/payloads/BookingPayloads.java
package com.restfulbooker.payloads;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * Builders de payloads — équivalent Java de payloads/booking_payloads.py.
 */
public class BookingPayloads {

    public static Map<String, Object> createBooking() {
        return createBooking(Map.of());
    }

    /** Payload complet avec valeurs par défaut, surchargeables via {@code overrides}. */
    public static Map<String, Object> createBooking(Map<String, Object> overrides) {
        Map<String, Object> booking = new LinkedHashMap<>();
        booking.put("firstname", "Jim");
        booking.put("lastname", "Brown");
        booking.put("totalprice", 111);
        booking.put("depositpaid", true);

        Map<String, Object> dates = new LinkedHashMap<>();
        dates.put("checkin", "2025-01-01");
        dates.put("checkout", "2025-01-10");

        Map<String, Object> merged = new LinkedHashMap<>(overrides);
        if (merged.containsKey("checkin")) dates.put("checkin", merged.remove("checkin"));
        if (merged.containsKey("checkout")) dates.put("checkout", merged.remove("checkout"));
        booking.put("bookingdates", dates);
        booking.put("additionalneeds", "Breakfast");

        booking.putAll(merged);
        return booking;
    }

    public static Map<String, Object> createBookingWithoutField(String field) {
        Map<String, Object> booking = createBooking();
        booking.remove(field);
        return booking;
    }

    /** Payload sans le champ optionnel additionalneeds. */
    public static Map<String, Object> createBookingMinimal() {
        Map<String, Object> booking = new LinkedHashMap<>();
        booking.put("firstname", "Jim");
        booking.put("lastname", "Brown");
        booking.put("totalprice", 111);
        booking.put("depositpaid", true);
        Map<String, Object> dates = new LinkedHashMap<>();
        dates.put("checkin", "2025-01-01");
        dates.put("checkout", "2025-01-10");
        booking.put("bookingdates", dates);
        return booking;
    }

    public static Map<String, Object> updateBooking(Map<String, Object> overrides) {
        return createBooking(overrides);
    }

    public static Map<String, Object> patchBooking(String firstname, String lastname, Integer totalprice) {
        Map<String, Object> payload = new LinkedHashMap<>();
        if (firstname != null) payload.put("firstname", firstname);
        if (lastname != null) payload.put("lastname", lastname);
        if (totalprice != null) payload.put("totalprice", totalprice);
        return payload;
    }

    public static Map<String, Object> authPayload(String username, String password) {
        Map<String, Object> payload = new LinkedHashMap<>();
        payload.put("username", username != null ? username : "admin");
        payload.put("password", password != null ? password : "password123");
        return payload;
    }
}
